### DISCUSSION TITLE
how to get element coordinate ?

### DISCUSSION THREAD

**Discussion by asevenm** [diegomura/react-pdf#2819] [Q&A]
Is there any methods to get element's coordinate. (like getBoundingClientRect)

**Comment by SerchoD** [comment]
Have you solved it?

I want to know the coordinates of row's table generating, to detect when the page break comes, to make separation...

**Comment by diegomura** [comment]
Hi! There is no public method like `getBoundingClientRect`, but the information does exist.

When react-pdf lays out your document, it computes a box for every element with its final `top`, `left`, `width` and `height`. You can get access to that resolved layout tree through the `onRender` prop on `Document`:

```jsx
<Document
  onRender={({ _INTERNAL__LAYOUT__DATA_ }) => {
    console.log(_INTERNAL__LAYOUT__DATA_);
  }}
>
```

Each node in that tree has a `box` property with the coordinates, and children nested the same way as your components. Note this data is available after rendering, not during it, and the property is internal, so its shape can change between versions.

If you need the coordinates without producing a PDF, you can also run the layout step directly with the `@react-pdf/layout` package and walk the object it returns.

Hope that helps!
